Automattic/wp-calypso
# Plugins: remove the legacy plugins-wpcom directory

The legacy UI that shows WP.com features on a `/plugins/manage` for Simple sites is no longer needed. We now have this functionality as part of the Plugins Browser UI (see #18323). Let's remove the `plugins-wpcom` directory now.

This is how the legacy UI looks like, just as a reminder:
<img width="569" alt="plugins-legacy-ui" src="https://user-images.githubusercontent.com/664258/31081784-3df4c788-a78d-11e7-8724-50eca9f0b86c.png">


The first commit redirects `/plugins/manage` for Simple sites to `/plugins` and removes the no-longer-needed `/plugins/category/:category/:site_id?` route. After this, there's no longer any route that would direct to the legacy WP.com UI.

The second commit removes rendering of the `WpcomPluginPanel` component as a special case for Simple sites. After this, the `WpcomPluginPanel` component, together with most of the `plugins-wpcom` directory, is no longer imported anywhere.

The third and fourth commit move two modules that will still be needed out of the `plugins-wpcom` directory:
- the `WpcomPluginInstallButton` that is rendered on a plugin page for a Simple site. It's the version of the button that initiates the Automated Transfer for the site. Moved to `plugins/plugin-install-button-wpcom`
- the data about WP.com features used to display them as Jetpack pseudo-plugins. This continues to be used by the features-as-plugins code from #18323. Just move it to a better location.

The last commit is the actual big delete: removes all remaining code from `plugins-wpcom`.

**How to test:**
1. Test that `/plugins/manage/:simpleSite` redirects to `/plugins/:simpleSite`
2. Test that `/plugins/category/engagement/:simpleSite` displays nothing.
3. Test that showing WP.com feature as plugins from #18323 still works: go to `/plugins/:simpleSite` and click the "Engagement" predefined search. Does it show Jetpack features as the first few results?
4. Get a Business site that's not Atomic yet and go to `/plugins/woocommerce`. Check that the "Install" button is there and that it triggers the AT. If you feel too lazy to do the AT, just check in React devtools  that the `WpcomPluginInstallButton` component is rendered there.
